front/migrations/20230614_timetttamp.ts

```diff
@@ -0,0 +1,74 @@
+/* Following a typo on points payload in our Qdrant database in which the field
+time*st*amp was written time*t*amp This migration creates a new field for every
+point, correctly named time*st*amp and with the value of the existing time*t*amp
+field*/
+
+import { QdrantClient } from "@qdrant/js-client-rest";
+
+// if reusing, set env vars to those values, don't hardcode the values in the code
+// also as a safety delete the env vars (if persisted one way or another) after the migration
+const { QDRANT_PROD_URL, QDRANT_PROD_API_KEY } = process.env;
+
+const prodClient = new QdrantClient({
+  url: QDRANT_PROD_URL,
+  apiKey: QDRANT_PROD_API_KEY,
+});
+
+async function migrateCollection(
+  client: QdrantClient,
+  collectionName: string,
+  batchSize = 250,
+  offset?: string | number | undefined | null | Record<string, unknown>
+) {
+  let currentTime = Date.now(),
+    points,
+    updated = 0;
+  let nb_points = await client.count(collectionName);
+  console.log(
+    "Migrating Collection ",
+    collectionName,
+    "\nNumber of points: ",
+    nb_points.count
+  );
+  // scroll points excluding those with timestamp field until none are left
+  do {
+    console.log("Current offset: ", offset);
+    const { points, next_page_offset } = await client.scroll(collectionName, {
+      limit: batchSize,
+      offset,
+      with_vector: false,
+      filter: {
+        must: [{ is_empty: { key: "timestamp" } }],
+      },
+    });
+
+    // update the points
+    const updatePromises = points.map(async (point) => {
+      const payload = { timestamp: point.payload?.timetamp };
```

Comment:
let's skip if timestamp is already there and console.log wether we skipped or actually did something